### DISCUSSION TITLE
辅筛模式能力优化建议

### DISCUSSION THREAD

**Discussion by skywb** [rimeinn/rime-moran#98] [Ideas]
仅使用音码作为输入， 辅助码仅存在最后位置。即每个字固定两个音码组成，辅助码用于筛选

1. 单字场景 

   单字场景为输出单字的场景， 单字场景一般是1码、2码、3码、4码。这种场景下可以通过辅助码进行首选筛选。

   功能列表： 

   - 输入辅助码，首选后移

     举例来讲， 当输入 `he` 时， 首选为 为 `喝`  时， 再次键入 `k` 时， `喝` 应后移让出首选位置， 可能变成 `呵`.   因为键入辅助码时意味着首选并不是我需要的，所以让出首选位置。

   - 四码单字优先

     四码单字优先模式用于界定输入四码时首选为单字场景，后两码作为辅助码

   

2. 词汇场景

   词汇场景为输出的字小于等于4的场景， 即码长小于10 （包括了最后一位辅助码）

   功能列表：

   - 输入辅助码首选后移

   - 四码词组优先

     输入四码时优先组成词汇，此时只能使用一个辅助码

   - 辅助码作用域

     - 词中首字
     - 词中尾字
     - 词中任意字

3. 整句场景

   整句场景为码长大于9时的场景，整句场景下，首选相对固定，只有一个辅助码生效，且辅助码作用于最后一个字

   功能列表：

   - 直接辅助码

     可以使用一个辅助码作用于最后一个字

   - tab候选模式

     键入tab时进入候选模式，输入辅助码作用于首字， 且上屏后清空辅助码， 再进行下一个字的候选

**Comment by liangkaichun** [comment]
1、辅筛模式是以词组输入为核心的，单字+两辅码，需要和四码双字词组调频；
2、整句输入，可以使用魔然整句；
3、辅筛模式下，节奏感很重要；
